### ISSUE TITLE
[BUG] No covers are displayed

### ISSUE THREAD

**Issue by Porkepix** [FoedusProgramme/Gramophone#37]
**Describe the bug**
Gramophone currently shows no covers at all

**To Reproduce**
Just install Gramophone, probably on older Android (see below) and have songs that should have covers whether it is in metadata or a cover picture in the folder and see that none are displayed

**Expected behavior**
Covers should be loaded

**Smartphone (please complete the following information):**
 - Device: Xperia XZ
 - OS: Android Oreo (8.0)
 - Version 1.0.1

**Additional context**
This might be because of some changes in behavior along Android versions, and maybe some information can be found there: https://github.com/chr56/Phonograph_Plus/issues/65#issuecomment-1794961758

These logs seems related:

<details>
<summary>
Logs
</summary>

```
11-06 17:41:03.324  3226  3426 E DatabaseUtils: Writing exception to parcel
11-06 17:41:03.324  3226  3426 E DatabaseUtils: java.lang.SecurityException: External path: /storage/emulated/0/Android/data/com.android.providers.media/albumthumbs/1620562783893: Neither user 10248 nor current process has android.permission.WRITE_EXTERNAL_STORAGE.
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.app.ContextImpl.enforce(ContextImpl.java:1735)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:1767)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:756)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.enforceCallingOrSelfPermissionAndAppOps(MediaProvider.java:5680)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.checkAccess(MediaProvider.java:4604)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.ensureFileExists(MediaProvider.java:3473)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getAlbumArtOutputUri(MediaProvider.java:4887)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.makeThumbInternal(MediaProvider.java:5020)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getThumb(MediaProvider.java:4950)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.openFile(MediaProvider.java:4503)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContentProvider.openAssetFile(ContentProvider.java:1545)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1725)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1791)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:425)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:308)
11-06 17:41:03.324  3226  3426 E DatabaseUtils: 	at android.os.Binder.execTransact(Binder.java:674)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: Writing exception to parcel
11-06 17:41:03.327  3226  3704 E DatabaseUtils: java.lang.SecurityException: External path: /storage/emulated/0/Android/data/com.android.providers.media/albumthumbs/1620562614225: Neither user 10248 nor current process has android.permission.WRITE_EXTERNAL_STORAGE.
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.app.ContextImpl.enforce(ContextImpl.java:1735)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:1767)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:756)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.enforceCallingOrSelfPermissionAndAppOps(MediaProvider.java:5680)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.checkAccess(MediaProvider.java:4604)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.ensureFileExists(MediaProvider.java:3473)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getAlbumArtOutputUri(MediaProvider.java:4887)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.makeThumbInternal(MediaProvider.java:5020)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getThumb(MediaProvider.java:4950)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.openFile(MediaProvider.java:4503)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContentProvider.openAssetFile(ContentProvider.java:1545)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1725)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1791)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:425)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:308)
11-06 17:41:03.327  3226  3704 E DatabaseUtils: 	at android.os.Binder.execTransact(Binder.java:674)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: Writing exception to parcel
11-06 17:41:03.340  3226  3240 E DatabaseUtils: java.lang.SecurityException: External path: /storage/emulated/0/Android/data/com.android.providers.media/albumthumbs/1611423074907: Neither user 10248 nor current process has android.permission.WRITE_EXTERNAL_STORAGE.
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.app.ContextImpl.enforce(ContextImpl.java:1735)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:1767)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:756)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.enforceCallingOrSelfPermissionAndAppOps(MediaProvider.java:5680)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.checkAccess(MediaProvider.java:4604)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.ensureFileExists(MediaProvider.java:3473)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getAlbumArtOutputUri(MediaProvider.java:4887)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.makeThumbInternal(MediaProvider.java:5020)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.getThumb(MediaProvider.java:4950)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at com.android.providers.media.MediaProvider.openFile(MediaProvider.java:4503)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContentProvider.openAssetFile(ContentProvider.java:1545)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1725)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContentProvider.openTypedAssetFile(ContentProvider.java:1791)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:425)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:308)
11-06 17:41:03.340  3226  3240 E DatabaseUtils: 	at android.os.Binder.execTransact(Binder.java:674)
```
</details>

What's strange though is that the application *does* have the storage permission though.

**Comment by nift4** [comment]
hi, please try: https://github.com/AkaneTan/Gramophone/files/13538499/Gramophone-1.0.2.bd8ec5-release.apk.zip

**Comment by Porkepix** [comment]
> hi, please try: [AkaneTan/Gramophone/files/13538499/Gramophone-1.0.2.bd8ec5-release.apk.zip](https://github.com/AkaneTan/Gramophone/files/13538499/Gramophone-1.0.2.bd8ec5-release.apk.zip)

I can confirm covers now displays.

Notes on that version though:
- Curiously by updating with this apk the icon that was placed on homescreen was removed in the process (dunno why?)
- Most settings menu can be slided too far while being very short but some of them such as Player UI can't be enough and therefore the end of it can't be completely read; I don't remember if it was the case before.
- And little personal preference for the previous icon where the vinyl player was more obvious I think :p (but maybe there's some consistency choice behind it, or it's to match Google's style guides).

**Comment by nift4** [comment]
please create new issue for settings including screenshots, they have been redesigned recently. the icon was changed because most people did not recognize it as gramophone - the homescreen icon disappearing can't be controlled by the app, it may be related to your launcher settings.

**Comment by Porkepix** [comment]
> please create new issue for settings including screenshots, they have been redesigned recently. the icon was changed because most people did not recognize it as gramophone - the homescreen icon disappearing can't be controlled by the app, it may be related to your launcher settings.

Well, my launcher is the default one and have no setting for this. I've already seen this before but only once on an application update, so I've a hunch it's a slight difference in a signature or something which leads the launcher to think the application related to the previous icon doesn't exist anymore, but I'm not really sure about this.
I could try to see if I can reproduce it and if there's something in logs, but I've the feeling I'm hoping for too much here.
And… I guess icons aren't recognized as well depending on the people, then.

**Comment by nift4** [comment]
seeing how major the changes from 1.0.1 to the current development version were, I'll write this down as a one-off issue; let me know if it happens again

**Comment by Porkepix** [comment]
Curiously, I went back to previous release after uninstalling the zip from here to check on other issues, and it asked for permission and did show me the covers. Pretty curious.
